File: Teste-IBBI-Backend/products/models/products_model.py
from sqlalchemy import Column, Integer, String, Numeric

from shared.database import Base


class ProductsM(Base):
    __tablename__ = 'products_data'
    productId = Column(Integer, primary_key=True, autoincrement=True)
    productName = Column(String(100))
    productImageUrl = Column(String(100))
    productPrice = Column(Numeric)
    productDesc = Column(String(500))
    productCategory = Column(String(40))
    productAmount = Column(Numeric)
    productAmountSuggestion = Column(Numeric)
    vendorName = Column(String(100), nullable=True)

    class Config:
        from_attributes = True

class ProductCategoriesM(Base):
    __tablename__ = 'product_categories'
    categoryId = Column(Integer, primary_key=True, autoincrement=True)
    categoryName = Column(String(80))
    categoryDesc = Column(String(500))

    class Config:
        from_attributes = True

class ProductSoldM(Base):
    __tablename__ = 'sold_products'
    id = Column(Integer, primary_key=True, autoincrement=True)
    clientName = Column(String(100), nullable=False)
    vendorName = Column(String(100), nullable=False)
    productId = Column(Integer)
    productName = Column(String(100))
    productCategory = Column(String(40))
    soldAmount = Column(Numeric)
    soldTime = Column(String)

    class Config:
        from_attributes = True

File: Teste-IBBI-Frontend/src/app/shared/models/products.model.ts
export interface ProductModel{
    productId?: number,
    productName: string,
    productImageUrl: string,
    productPrice: number,
    productDesc: string,
    productCategory: string,
    productAmount:number,
    productAmountSuggestion: number
    vendorName?: string
}

export interface CategoriesModel{
    categoryId?: number
    categoryName:string
    categoryDesc?: string
}

export interface BuyModel{
    productId?:number,
    productName:string,
    productCategory:string,
    soldAmount:number,
    clientName?:string,
    vendorName?:string,
    soldTime?:string
}